src/api/app/models/package.rb

```diff
@@ -397,7 +397,12 @@ def unlock_by_request(request)
 
   def sources_changed(opts = {})
     dir_xml = opts[:dir_xml]
-    update_activity
+
+    # to call update_activity before filter
+    # NOTE: We need `Time.now`, otherwise the old tests suite doesn't work,
+    # remove it when removing the tests
+    update({updated_at: Time.now})
```

Comment:
Yes, it is for that, and it shouldn't be like that. That's why I added the comments. It was already like that as we were doing this here: https://github.com/openSUSE/open-build-service/pull/2777/files/6389055e818a7c693fb9c7142398f0b1ef30a129#diff-37dd923f1e6421bf4aa8375fbe11ad01L830

and this is also done here: https://github.com/openSUSE/open-build-service/pull/2777/files/6389055e818a7c693fb9c7142398f0b1ef30a129#diff-37dd923f1e6421bf4aa8375fbe11ad01R810

I think it is better to remove it when we remove the old test suite that trying to fix it now.